File: backend/src/main/java/com/chenxiang/biotree/domain/taxon/TaxonView.java
/**
 * 分类树展示视图：简易七级 / 完整阶元。
 *
 * Author: chen-xiang
 * Created: 2026-08-31
 */
package com.chenxiang.biotree.domain.taxon;

import java.util.Locale;

public enum TaxonView {
    SIMPLE,
    FULL;

    public static TaxonView fromParam(String raw) {
        if (raw == null || raw.isBlank()) {
            return SIMPLE;
        }
        return switch (raw.trim().toLowerCase(Locale.ROOT)) {
            case "full" -> FULL;
            default -> SIMPLE;
        };
    }
}

File: frontend/src/stores/taxonView.ts
/**
 * 分类树展示视图偏好（简易七级 / 完整阶元）。
 *
 * Author: chen-xiang
 * Created: 2026-08-31
 */
import { defineStore } from 'pinia'
import { ref, watch } from 'vue'
import type { TaxonView } from '@/api/taxon'

const STORAGE_KEY = 'biotree.taxonView'

function readInitial(): TaxonView {
  const raw = localStorage.getItem(STORAGE_KEY)
  return raw === 'full' ? 'full' : 'simple'
}

export const useTaxonViewStore = defineStore('taxonView', () => {
  const view = ref<TaxonView>(readInitial())

  watch(view, (v) => {
    localStorage.setItem(STORAGE_KEY, v)
  })

  function setView(next: TaxonView) {
    view.value = next
  }

  function toggle() {
    view.value = view.value === 'simple' ? 'full' : 'simple'
  }

  return { view, setView, toggle }
})
